ui/button.js

```diff
@@ -56,355 +39,267 @@ $.widget( "ui.button", {
+		if ( key === "icon" ) {
+			if ( value !== null ) {
+				this._updateIcon( value );
+			} else {
+				this.icon.remove();
+				this._removeClass( "ui-button-icon", " ui-icon-" + this.options.iconPosition );
+			}
+		}
+
+		// Make sure we can't end up with a button that has no text nor icon
+		if ( key === "showLabel" ) {
+			if ( ( !value && this.options.icon ) || value ) {
+				this._toggleClass( this._classes( "ui-button-icon-only" ), null, !value )
+					._toggleClass( null, this.options.iconPosition, value );
+				this._updateTooltip();
+			} else {
+				value = true;
+			}
+		}
+		if ( key === "iconPosition" && this.options.icon ) {
+			this._addClass( null, value )
+				._removeClass( null, this.options.iconPosition );
+		}
+		if ( key === "label" ) {
+			if ( this.isInput ) {
+				this.element.val( value );
+			} else {
+
+				// If there us an icon append it else nothing then append the value
+				// this avoids removal of the icon when setting label text
+				this.element.html( !!this.icon ? "" : this.icon ).append( value );
+			}
+		}
 		this._super( key, value );
 		if ( key === "disabled" ) {
-			this.widget().toggleClass( "ui-state-disabled", !!value );
-			this.element.prop( "disabled", !!value );
-			if ( value ) {
-				if ( this.type === "checkbox" || this.type === "radio" ) {
-					this.buttonElement.removeClass( "ui-state-focus" );
-				} else {
-					this.buttonElement.removeClass( "ui-state-focus ui-state-active" );
-				}
-			}
-			return;
+			this._toggleClass( null, "ui-state-disabled", value ).element[ 0 ].disabled = value;
+			this.element.blur();
 		}
-		this._resetButton();
 	},
 
 	refresh: function() {
-		//See #8237 & #8828
-		var isDisabled = this.element.is( "input, button" ) ? this.element.is( ":disabled" ) : this.element.hasClass( "ui-button-disabled" );
+
+		// Make sure to only check disabled if its an element that supports this otherwise
+		// check for the disabled class to determine state
+		var isDisabled = this.element.is( "input, button" ) ?
+			this.element[ 0 ].disabled : this.element.hasClass( "ui-button-disabled" );
 
 		if ( isDisabled !== this.options.disabled ) {
-			this._setOption( "disabled", isDisabled );
+			this._setOptions({ "disabled": isDisabled });
 		}
-		if ( this.type === "radio" ) {
-			radioGroup( this.element[0] ).each(function() {
-				if ( $( this ).is( ":checked" ) ) {
-					$( this ).button( "widget" )
-						.addClass( "ui-state-active" )
-						.attr( "aria-pressed", "true" );
+
+		this._updateTooltip();
+	}
+});
+
+// DEPRECATED
+if ( $.uiBackCompat !== false ) {
+
+	// Text and Icons options
+	$.widget( "ui.button", $.ui.button, {
+		options: {
+			text: true,
+			icons: {
+				primary: null,
+				secondary: null
+			}
+		},
+
+		_create: function() {
+			if ( this.options.showLabel && !this.options.text ) {
+				this.options.showLabel = this.options.text;
+			}
+			if ( !this.options.showLabel && this.options.text ) {
+				this.options.text = this.options.showLabel;
```

Comment:
The conditionals are checking if the values are actually different and have changed from the default. In the first case if `showLabel` is `true` and `text` is `false` that means that the `text` option has been changed so we update the `showLabel` option to match.

In the second one we check that `showLabel` is `false` and `text` is `true` then the `showLabel` option has been changed and we need to update `text` to match.